libs/datasets/timeseries.py

```diff
@@ -282,29 +296,55 @@ def load_csv(cls, path_or_buf: Union[pathlib.Path, TextIO]):
         return cls(df)
 
 
+def _add_location_id(df: pd.DataFrame):
+    """Adds the location_id column derived from FIPS, inplace."""
+    if CommonFields.LOCATION_ID in df.columns:
+        raise ValueError("location_id already in DataFrame")
+    df[CommonFields.LOCATION_ID] = df[CommonFields.FIPS].apply(pipeline.fips_to_location_id)
+
+
+def _add_fips_if_missing(df: pd.DataFrame):
+    """Adds the FIPS column derived from location_id, inplace."""
+    if CommonFields.FIPS not in df.columns:
+        df[CommonFields.FIPS] = df[CommonFields.LOCATION_ID].apply(pipeline.location_id_to_fips)
+
+
 @final
 @dataclass(frozen=True)
-class MultiRegionTimeseriesDataset:
+class MultiRegionTimeseriesDataset(SaveableDatasetInterface):
     """A set of timeseries with values from any number of regions."""
 
     # `data` may be used to process every row without considering the date or region. Keep logic about
-    # FIPS/locationID/region containing in this class by using methods such as `get_one_region`. Do *not*
-    # read date or region related columns directly from `data`.
+    # FIPS/location_id/region containing in this class by using methods such as `get_one_region`. Do
+    # *not* read date or region related columns directly from `data`.
     data: pd.DataFrame
 
     provenance: Optional[pd.Series] = None
 
+    @property
+    def dataset_type(self) -> DatasetType:
+        return DatasetType.MULTI_REGION
+
+    @classmethod
+    def load_csv(cls, path_or_buf: Union[pathlib.Path, TextIO]):
+        return MultiRegionTimeseriesDataset.from_csv(path_or_buf)
+
     @staticmethod
     def from_csv(path_or_buf: Union[pathlib.Path, TextIO]) -> "MultiRegionTimeseriesDataset":
         df = common_df.read_csv(path_or_buf, set_index=False)
+        if CommonFields.LOCATION_ID not in df.columns:
+            raise ValueError("MultiRegionTimeseriesDataset.from_csv requires location_id column")
+        _add_fips_if_missing(df)
         return MultiRegionTimeseriesDataset(df)
 
     @staticmethod
     def from_timeseries(ts: TimeseriesDataset) -> "MultiRegionTimeseriesDataset":
-        return MultiRegionTimeseriesDataset(ts.data, provenance=ts.provenance)
+        df = ts.data.copy()
+        _add_location_id(df)
+        return MultiRegionTimeseriesDataset(df, provenance=ts.provenance)
```

Comment:
Should provenance be by location id too?

Reply:
Good point. Added test and fixed.